akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala

```diff
@@ -646,6 +646,14 @@ object PersistentActorSpec {
     }
   }
 
+  class PersistInRecovery(name: String) extends ExamplePersistentActor(name) {
+    override def receiveRecover = super.receiveRecover orElse {
+      case RecoveryCompleted ⇒
+        persist(Evt(RecoveryCompleted))(updateState)
```

Comment:
would also be interesting with a test that is using `persistAsync` from `receiveRecover`